Add intermediate solution to Algorithm-Roman-Numeral-Converter

mrthnmn opened the pull request:
A second (intermediate) solution has been added.

mrthnmn committed:
Added a new solution that others might find interesting.

mrthnmn committed:
Removed my username from credits.

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -68,6 +68,70 @@ convertToRoman(36);
 - [For Loops](JS-For-Loop)
 - [While Loops](JS-while-Loop)
 
+## :intermediate: Intermediate Code Solution:
```
~~`:intermediate:`~~ `:sunflower:`

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -68,6 +68,70 @@ convertToRoman(36);
+
+:rocket: [Run Code](https://repl.it/C1YV)
+
+## Code Explanation:
```
~~`## Code Explanation:`~~ `### Code Explanation:`

Rafase282 commented:
Please add a section to our articles related to your solution.

mrthnmn commented:
I'm afraid I'm not sure what you want. I thought I was editing the existing
wiki article on the Roman Numeral Converter Algorithm challenge. Do you
want a more-thorough explanation of the code I added or a completely new
wiki article?

On Thu, Jun 16, 2016 at 12:18 PM, Rafael J. Rodriguez <
notifications@github.com> wrote:

> Please add a section to our articles related to your solution.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FreeCodeCamp/wiki/pull/1151#issuecomment-226536311>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AQKaMFz1fpkkEt55GlFjrDTruTymn_pPks5qMXdGgaJpZM4I3gRi>
> .
>

Rafase282 commented:
Please take a look here https://github.com/FreeCodeCamp/wiki/issues/1094

We have a template for solutions, and you adding a new one also requires a section with related links. I was asking to add it and use links form our wiki if possible. Sorry for not being clear before.

mrthnmn committed:
- Add links after the problem explanation
- Add links after both code explanations
- Add a sunflower image in front of "Intermediate Code Solution"
- Clean up code in Intermediate Code Solution
- Add "@mrthnmn" to credits

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -6,6 +6,13 @@
+
+- [Roman Numerals](http://www.mathsisfun.com/roman-numerals.html)
+- [Array.splice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
+- Array.indexOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
```
~~`Array.indexOf()]`~~ `[Array.indexOf()]`

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
 
-- [For Loops](JS-For-Loop)
-- [While Loops](JS-while-Loop)
+- [For Loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for)
```
Change this so it links to our internal article: [`JS-For-Loop`](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md).

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
-- [For Loops](JS-For-Loop)
-- [While Loops](JS-while-Loop)
+- [For Loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for)
+- [While Loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/while)
```
Change this so it links to our internal article: [`JS-While-Loop`](https://github.com/FreeCodeCamp/wiki/blob/master/JS-While-Loop.md).

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -6,6 +6,13 @@
+#### Relevant Links
+
+- [Roman Numerals](http://www.mathsisfun.com/roman-numerals.html)
+- [Array.splice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
```
Change this so it links to our internal article: [`JS-Array-Prototype-Splice`](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Splice.md).

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -6,6 +6,13 @@
+- [Roman Numerals](http://www.mathsisfun.com/roman-numerals.html)
+- [Array.splice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
+- Array.indexOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
+- [Array.join()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join)
```
Change this so it links to our internal article: [`JS-Array-Prototype-Join`](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Join.md).

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
+
+#### Relevant Links
+
+- [For Loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for)
```
Change this so it links to our internal article: [`JS-For-Loop`](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md).

atjonathan commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
+#### Relevant Links
+
+- [For Loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for)
+- [While Loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/while)
```
Change this so it links to our internal article: [`JS-While-Loop`](https://github.com/FreeCodeCamp/wiki/blob/master/JS-While-Loop.md).

mrthnmn committed:
- Change links from MDN articles to internal FreeCodeCamp articles, where available.

Rafase282 commented:
Please try to fix the conflicts on your branch. Feel free to come to the wiki chat if you need help.

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -6,6 +6,13 @@
+
+- [Roman Numerals](http://www.mathsisfun.com/roman-numerals.html)
+- [Array.splice()](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Splice.md)
+- Array.indexOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
```
```diff
-- Array.indexOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
+- [Array.indexOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
```

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -6,6 +6,13 @@
+#### Relevant Links
+
+- [Roman Numerals](http://www.mathsisfun.com/roman-numerals.html)
+- [Array.splice()](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Splice.md)
```
```diff
-- [Array.splice()](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Splice.md)
+- [Array.splice()](JS-Array-Prototype-Splice)
```

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -6,6 +6,13 @@
+- [Roman Numerals](http://www.mathsisfun.com/roman-numerals.html)
+- [Array.splice()](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Splice.md)
+- Array.indexOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
+- [Array.join()](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Join.md)
```
```diff
-- [Array.join()](https://github.com/FreeCodeCamp/wiki/blob/master/JS-Array-Prototype-Join.md)
+- [Array.join()](JS-Array-Prototype-Join)
```

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
 
-- [For Loops](JS-For-Loop)
-- [While Loops](JS-while-Loop)
+- [For Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md)
```
```diff
-- [For Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md)
+- [For Loops](JS-For-Loop)
```

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
-- [For Loops](JS-For-Loop)
-- [While Loops](JS-while-Loop)
+- [For Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md)
+- [While Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-While-Loop.md)
```
```diff
-- [While Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-While-Loop.md)
+- [While Loops](JS-While-Loop)
```

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
+
+#### Relevant Links
+
+- [For Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md)
```
```diff
-- [For Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md)
+- [For Loops](JS-For-Loop)
```

Rafase282 commented on Algorithm-Roman-Numeral-Converter.md:
```diff
@@ -65,14 +72,83 @@ convertToRoman(36);
+#### Relevant Links
+
+- [For Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-For-Loop.md)
+- [While Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-While-Loop.md)
```
```diff
-- [While Loops](https://github.com/FreeCodeCamp/wiki/blob/master/JS-While-Loop.md)
+- [While Loops](JS-While-Loop)
```

atjonathan commented:
@mrthnmn, please resolve merge conflicts and implement suggestions :wink:

mrthnmn commented:
I will do on Monday. I wasn't able to on Saturday because I was behind a
corporate proxy.

On Sat, Jun 18, 2016 at 2:27 PM, Jonathan <notifications@github.com> wrote:

> @mrthnmn <https://github.com/mrthnmn>, please resolve merge conflicts and
> implement suggestions 😉
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FreeCodeCamp/wiki/pull/1151#issuecomment-226957672>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AQKaMHK8qlsdYSabI93g0N0J4BV1uwMcks5qNDiEgaJpZM4I3gRi>
> .
>

alayek commented:
@mrthnmn this branch has conflicts. You might need to rebase the branch.

mrthnmn commented:
Attempt to resolve conflicts
